bin/specextract

```diff
@@ -79,381 +81,449 @@ v4 = make_verbose_level(toolname, 4)
+    For mkrmf: input FEF file and no weights file and return RMF name.
+    For mkacisrmf: input CALDB and WMAP='none' and return RMF name.
+    
+    Returns unweighted ACIS RMF filename.
     """
 
     v1(infostr)
     
     rmffile = f"{full_outroot}.rmf"
 
-    if "mkrmf" == rmftool:
-
-        mkrmf.punlearn()
+    try:
+        with new_pfiles_environment(ardlib=True,copyuser=False):
+            if rmftool == "mkrmf":
+                mkrmf.infile = feffile
+                mkrmf.outfile = rmffile
+                mkrmf.logfile = ""
+                mkrmf.weights = ""
+                mkrmf.axis1 = f"energy={ebin}"
+                mkrmf.axis2 = rmfbin
+                mkrmf.clobber = clobber
+                mkrmf.verbose = str(verbose)
+
+                mkrmf()
 
-        mkrmf.infile = "CALDB"
-        mkrmf.outfile = rmffile
-        mkrmf.logfile = ""
-        mkrmf.weights = weightfile
-        mkrmf.axis1 = f"energy={ebin}"
-        mkrmf.axis2 = rmfbin
-        mkrmf.clobber = clobber
-        mkrmf.verbose = str(verbose)
+            else:
+                channel = rmfbin.split("=")
+                channel.reverse()
+
+                mkacisrmf.punlearn()
+
+                # force CALDB querry to match 'ccd_id' parameter, otherwise the default
+                # behavior is to use the CCD_ID header keyword in the 'obsfile'
+                mkacisrmf.infile = f"CALDB(CCD_ID={ccd_id})"
+                mkacisrmf.outfile = rmffile
+                mkacisrmf.energy = ebin
+                mkacisrmf.channel = channel[0]
+                mkacisrmf.chantype = ptype.upper()
+                mkacisrmf.wmap = "none"
+                mkacisrmf.gain = "CALDB"
+                mkacisrmf.obsfile = evt_filename
+                mkacisrmf.ccd_id = ccd_id
+                mkacisrmf.chipx = chipx
+                mkacisrmf.chipy = chipy
+                mkacisrmf.clobber = clobber
+                mkacisrmf.verbose = str(verbose)
+
+                mkacisrmf()
 
-        mkrmf()
-        
         return rmffile
 
-    elif "mkacisrmf" == rmftool:
-
-        channel = rmfbin.split("=")
-        channel.reverse()
-
-        mkacisrmf.punlearn()
-
-        mkacisrmf.infile = "CALDB"
-        mkacisrmf.outfile = rmffile
-        mkacisrmf.energy = ebin
-        mkacisrmf.channel = channel[0]
-        mkacisrmf.chantype = ptype.upper()
-        mkacisrmf.gain = "CALDB"
-        mkacisrmf.clobber = clobber
-        mkacisrmf.verbose = str(verbose)
-        mkacisrmf.ccd_id = "0" # hopefully, this par will
-                               # be ignored since it can't
-                               # be set to "".
-        if wmap_clip:
-            mkacisrmf.wmap = f"{wmap_sky2tdet}[wmap]"
-        else:
-            mkacisrmf.wmap = f"{specfile}[WMAP]" # dmextract WMAP is faster than sky2tdet WMAP
-
-        mkacisrmf()
-        
-        return rmffile
-        
-    else:
+    except (OSError,IOError):
```

Comment:
`IOError` is the same as `OSError` in Python 3, so this can just be

```suggestion
    except OSError:
```

Reply:
good to know.  I've revised this on my local version.